cmd/timescale-prometheus/main.go

```diff
@@ -3,265 +3,34 @@
 // LICENSE for a copy of the license.
 package main
 
-// Based on the Prometheus remote storage example:
-// documentation/examples/remote_storage/remote_storage_adapter/main.go
-
 import (
-	"context"
 	"flag"
 	"fmt"
-	"net/http"
-	pprof "net/http/pprof"
 	"os"
-	"regexp"
-	"strings"
-	"sync/atomic"
-	"time"
 
-	"github.com/jackc/pgx/v4/pgxpool"
-	_ "github.com/jackc/pgx/v4/stdlib"
-	"github.com/jamiealquiza/envy"
-	"github.com/prometheus/client_golang/prometheus/promhttp"
-	"github.com/timescale/timescale-prometheus/pkg/api"
 	"github.com/timescale/timescale-prometheus/pkg/log"
-	"github.com/timescale/timescale-prometheus/pkg/pgclient"
-	"github.com/timescale/timescale-prometheus/pkg/pgmodel"
-	"github.com/timescale/timescale-prometheus/pkg/util"
+	"github.com/timescale/timescale-prometheus/pkg/timescale_prometheus"
 	"github.com/timescale/timescale-prometheus/pkg/version"
 )
 
-type config struct {
-	listenAddr        string
-	telemetryPath     string
-	pgmodelCfg        pgclient.Config
-	logLevel          string
-	haGroupLockID     int
-	restElection      bool
-	prometheusTimeout time.Duration
-	electionInterval  time.Duration
-	migrate           bool
-	stopAfterMigrate  bool
-	corsOrigin        *regexp.Regexp
-}
-
-const (
-	promLivenessCheck = time.Second
-)
-
-var (
-	elector *util.Elector
-)
-
 func main() {
-	cfg, err := parseFlags()
+	logLevel := flag.String("log-level", "debug", "The log level to use [ \"error\", \"warn\", \"info\", \"debug\" ].")
+	cfg := &timescale_prometheus.Config{}
+	cfg, err := timescale_prometheus.ParseFlags(cfg)
 	if err != nil {
 		fmt.Println("Version: ", version.Version, "Commit Hash: ", version.CommitHash)
 		fmt.Println("Fatal error: cannot parse flags ", err)
```

Comment:
Shouldn't we be using `Fprintf()` here (and below) with `os.Stderr` than going this way?

Reply:
This one is going to postponed to a later PR where we switch our default log format to match prometheus's, issue #261.